theano/sandbox/cuda/cuda_ndarray.cu

```diff
@@ -1064,9 +1064,12 @@ CudaNdarray_TakeFrom(CudaNdarray * self, PyObject *args){
         Py_DECREF(indices_obj);
         Py_DECREF(indices_float32);
     } else {
-        PyErr_SetString(PyExc_TypeError,
-                        "CudaNdarray_TakeFrom: need a CudaNdarray(float32) that"
-                        " is a view from int64 data for indices");
+        PyObject* py_s = PyObject_Str(indices_obj);
+        const char* s = PyString_AsString(py_s);
+        PyErr_Format(PyExc_TypeError,
+                     "CudaNdarray_TakeFrom: need an ndarray of int64 or a"
+                     " CudaNdarray(float32) that is a view from int64 data"
+                     " for indices. Got %s", s);
```

Comment:
I think `py_s` should be decrefed before returning, since `PyObject_Str` returns a fresh reference.